src/ooxml/java/org/apache/poi/xslf/usermodel/XSLFTextRun.java

```diff
@@ -586,10 +597,12 @@ void copy(XSLFTextRun r){
 
         PaintStyle srcFontColor = r.getFontColor();
         if(srcFontColor != null && !srcFontColor.equals(getFontColor())){
-            setFontColor(srcFontColor);
+            // setFontColor will throw IllegalArgumentException if this is not checked
+            if (srcFontColor instanceof SolidPaint)
+                setFontColor(srcFontColor);
         }
 
-        double srcFontSize = r.getFontSize();
+        Double srcFontSize=r.getFontSize();
```

Comment:
could you retain the original code formatting? ie whitespace around the '=' char

Reply:
I am sorry about the formatting - but this one of the key changes of the patch. Here you get a null pointer exception if getFontSize returns null